src/units/abilities.cpp

```diff
@@ -2333,6 +2333,8 @@ bool attack_type::has_ability_with_filter(const config & filter) const
 		return false;
 	}
 	const unit_map& units = get_unit_map();
+	bool check_adjacent = !filter.has_attribute("affect_adjacent") || filter["affect_adjacent"].to_bool();
+	bool check_distant = !filter.has_attribute("affect_distant") || filter["affect_distant"].to_bool();
```

Comment:
`!filter.has_attribute("affect_adjacent")` is redundent isn't it? Since `filter["affect_adjacent"].to_bool()` would return false if the attribute doesn't exist.

Reply:
By default chec must be done exxcept if affect_adjacent =no is specified